core/src/main/scala/com/avast/clients/rabbitmq/converters.scala

```diff
@@ -21,6 +21,8 @@ trait CheckedDeliveryConverter[A] extends DeliveryConverter[A] {
 
 object DeliveryConverter {
   implicit val identity: DeliveryConverter[Bytes] = (b: Bytes) => Right(b)
+  implicit val bytesArray: DeliveryConverter[Array[Byte]] = (b: Bytes) => Right(b.toByteArray)
+  implicit val utf8String: DeliveryConverter[String] = (b: Bytes) => Right(b.toStringUtf8)
```

Comment:
I'm thinking whether you shouldn't turn this to `CheckedDeliveryConverter`. The difference would be that it would also be usable in `MultiFormatConsumer` where _content type_ property is checked first before using the converter. Check `JsonDeliveryConverter` for example.

Reply:
I was also thinking about this but it's tricky to check the Content-Type correctly :( E.g. we could accept all `text/*` plus `application/json` but it would be very opinionated.

Reply:
I'd stick with `text/plain`, but `text/*` seems good too - it's a text. I'd skip the JSON, because it's be the same problem with many other formats which are in-fact text (YAML, XML, CSV, ...?).  
@petr-k @jakubjanecek - your opinion on this?

Reply:
I think supporting `text/*` seems reasonable.

Reply:
Second this.

On a related note: we're assuming UTF-8 here, which might not always be the correct choice. We might want to extract the encoding from the `Content-Type` attribute (if present, otherwise fallback to UTF-8).

Reply:
I would prefer not to make any `properties` changes in these _basic_ converters.

The reason is that `application/octet-stream` or `text/plain` are actually _super-types_ for more content types (like `application/json`). E.g. you want to generate a JSON and you cannot use circe so you will use the `utf8String` or `bytes` converter. And if you don't set the `Content-Type` explicitly then it will be automatically set to `text/plain` or `application/octet-stream`, which could be misleading for the consumer.

In our case, the converter used in consumer could refuse a message with `plain/text` (when looking for json) but would process a message without `Content-Type`.

Reply:
I believe this is very common behavior for client/server libraries nowadays - if you pass `String`, it's sent as `text/plain` except you say something else explicitly. The difference is that HTTP request almost always contains that header (it's not required by RFC, but it's both highly recommended and used) while it's quite common do not sent this property in Rabbit (at least for us).  
I don't understand the last paragraph - why do you think it would "process a message without Content-Type"? In case of `MultiFormatConsumer`, it would not. In case of common consumer, it would (because there's not Content-Type check since it doesn't make sense there - or does it? :thinking_face:) indeed, there's not other choice.

Reply:
I've never used `MultiFormatConsumer` so yes, I'm talking about common consumers.

Let me repeat my reason in different words: Now, we are publishing many events without `Content-Type`, and consumers don't require `Content-Type` - they just try to parse the content.

If we added `Content-Type` to every published message at this level (like generic publisher for `Array[Byte]`), then all published messages would have `application/octet-stream` and so could be refused by some consumer (because it checks if `Content-Type` exists and if so then it must be `application/json`).

So I see this peril in this change. We must ensure that after upgrading to _this_ version, we must ensure that correct `Content-Type` is set for all published events.

Reply:
Btw. I'm not against it - I will implement it in few minutes :) I just wanted to say that we should be careful :)

Reply:
Discussed in person - common `Consumer` does **not** do any check before the conversion thus this will not break anything.